From a086cc1a5c3cc75ff5b92cae8594079f3f107503 Mon Sep 17 00:00:00 2001
From: Ricardo Cerqueira <cyanogenmod@cerqueira.org>
Date: Mon, 29 Jul 2013 23:45:45 +0100
Subject: [PATCH] libgui: Add pre-MR2-compatible symbol for setPosition

SurfaceControl::setPosition now takes floats instead of integers as
arguments, which breaks linkability from pre-MR2 blobs. Add a direct
compatibility wrapper, since the values are treated in the same
way.

Change-Id: I293d117b94220258a7dbe354d6c0ee26a0d4314a

Conflicts:
	libs/gui/SurfaceControl.cpp
---
 libs/gui/SurfaceControl.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libs/gui/SurfaceControl.cpp b/libs/gui/SurfaceControl.cpp
index dc5fd33..6731543 100644
--- a/libs/gui/SurfaceControl.cpp
+++ b/libs/gui/SurfaceControl.cpp
@@ -105,10 +105,12 @@ status_t SurfaceControl::setPosition(float x, float y) {
     return mClient->setPosition(mHandle, x, y);
 }
 
+#ifdef ADD_LEGACY_SET_POSITION_SYMBOL
 extern "C" int _ZN7android14SurfaceControl11setPositionEff(float x, float y);
 extern "C" int _ZN7android14SurfaceControl11setPositionEii(int32_t x, int32_t y) {
     return _ZN7android14SurfaceControl11setPositionEff((float)x, (float)y);
 }
+#endif
 
 status_t SurfaceControl::setSize(uint32_t w, uint32_t h) {
     status_t err = validate();
-- 
1.9.1

